src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2104,6 +2124,17 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                             break;
                     }
                 }
+                else if (boundExpr is BoundConversion { ConversionKind: ConversionKind.MethodGroup, Type: { TypeKind: TypeKind.FunctionPointer }, SymbolOpt: var symbol })
+                {
+                    // Because the method group is a separate syntax node from the &, the lowest bound node here is the BoundConversion. However,
+                    // the conversion represents an implicit method group conversion from a typeless method group to a function pointer type, so
+                    // we should reflect that in the types and conversion we return.
+                    convertedType = type;
+                    convertedNullability = nullability;
+                    conversion = new Conversion(ConversionKind.MethodGroup, symbol, isExtensionMethod: false);
```

Comment:
>conversion = new Conversion(ConversionKind.MethodGroup, symbol, isExtensionMethod: false); [](start = 20, length = 90)

Could you point me to a test that covers this code path?

Reply:
One example would be `PublicApi_SemanticInfo01` (`FunctionPointerTypeSymbolTests.cs` line 1585). Basically, we want to make `&M` represent the language semantic here: it's a typeless expression that can be target-typed via an implicit MethodGroup conversion.

---
In reply to: [430778595](https://github.com/dotnet/roslyn/pull/44436#discussion_r430778595) [](ancestors = 430778595)

Reply:
> One example would be PublicApi_SemanticInfo01 (FunctionPointerTypeSymbolTests.cs line 1585). Basically, we want to make &M represent the language semantic here: it's a typeless expression that can be target-typed via an implicit MethodGroup conversion.

It doesn't look like we are asserting conversion information there.

---
In reply to: [430781925](https://github.com/dotnet/roslyn/pull/44436#discussion_r430781925) [](ancestors = 430781925,430778595)

Reply:
Added to `VerifyFunctionPointerSemanticInfo` for all address of expressions.

---
In reply to: [430850328](https://github.com/dotnet/roslyn/pull/44436#discussion_r430850328) [](ancestors = 430850328,430781925,430778595)